django/contrib/gis/db/models/functions.py

```diff
@@ -87,6 +90,8 @@ def as_postgresql(self, compiler, connection):
 
 class GeoFuncWithGeoParam(GeoFunc):
     def __init__(self, expression, geom, *expressions, **extra):
+        if not isinstance(geom, Geometry):
+            raise TypeError("Please provide a geometry object.")
```

Comment:
Maybe it's a separate issue (or no issue at all, since this message is similar to the existing one), but as a non-GIS user, this error doesn't make it obvious to me where the problem lies. It could be tougher in a more complex queryset also. What do you think?
```
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/django/tests/gis_tests/rasterapp/test_rasterfield.py", line 339, in test_db_function_errors
    RasterModel.objects.annotate(distance_from_point=Distance("geom", rast))
  File "/home/tim/code/django/django/contrib/gis/db/models/functions.py", line 239, in __init__
    super(Distance, self).__init__(*expressions, **extra)
  File "/home/tim/code/django/django/contrib/gis/db/models/functions.py", line 94, in __init__
    raise TypeError("Please provide a geometry object.")
TypeError: Please provide a geometry object.
```